File: leetcode/Python/complement-of-base-10-integer.py
class Solution:
    def bitwiseComplement(self, n: int) -> int:
        tot = 1<<30
        if n == 0:
            return 1
        while tot>0:
            if tot & n > 0:
                break
            tot>>=1
        tot = (tot<<1) - 1
        return tot^n

File: leetcode/C++/complement-of-base-10-integer.cpp
class Solution {
public:
    int bitwiseComplement(int n) {
        int tot=1<<30;
        if(n==0) return 1;
        for(; tot>0; tot>>=1) if(tot&n) break;
        tot=(tot<<1)-1;
        return tot^n;
    }
};